docs/DeveloperGuide.md

```diff
@@ -1,34 +1,173 @@
+### 1.3 Scope
+The intended audience of this document is the developers, designers, and software testers of CEGMods.
 
-{Describe the value proposition: what problem does it solve?}
+## 2. Setting up and getting started
+Setting up the project in your computer
+> <font color=#FF5733>Caution: </font> <font color=#00000><font color=#00000>  **Please follow each steps in the guide carefully.** </font>
 
-## User Stories
+#### Before you start:
+Please ensure you have Java 11 installed in you computer.
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+1. Fork this repo from this [link](https://github.com/AY2021S1-CS2113T-F11-2/tp.git)
+2. Clone the fork to your own computer. You are highly recommended to use Git tool (like Sourcetree) to track your work. 
+3. Use your own IDEA to program. You are highly recommended to use IntelliJ IDEA. Please check whether you have correct JDK version (JDK 11) in your computer and congiure the JDK.
 
-## Non-Functional Requirements
+    If you are using IntelliJ IDEA: 
+    - Open IntelliJ
+    - Set up JDK 11 for Gradle
+        1. Click `Configure` -> `Project Defaults` -> `Project Structure`
+        2. Click `New...` and set it to the directory of the JDK 11 installed
+4. Import the project as a Gradle project.
+    > <font color=#FF5733>Note: </font> <font color=#00000> It is different from importing a normal Java project.</font>
 
-{Give non-functional requirements}
+    If you are using IntelliJ IDEA:
+    - Click `Import Project` (or `Open or Import` in newer version of Intellij).
+    - Locate the `build.gradle` file (not the root folder as you would do in a normal importing) and select it. Click `OK`.
+    - Click `OK` to accept the default settings but do ensure that the selected version of `Gradle JVM` matches the JDK being used for the project.
+    - Wait for importing process to complete
+5. Do the testing. Please follow the testing guide
+   
+## 3. Design 
+### 3.1 Architecture
+The ***Architecture diagram*** given below shows the major components, and explains the structure of the software system.
+[![Architecture diagram](https://iili.io/3Ei2Vf.md.png)](https://freeimage.host/i/3Ei2Vf)
 
-## Glossary
+### 3.2 Classes
 
-* *glossary item* - Definition
+The CEGMods consists of six classes:
+* `Main`
+* `Ui`
+* `Parser`
+* `Module`
+* `Task`
+* `TimeTable`
+
+#### 3.2.1 Ui class
+The Ui class reads the uses's input and response relevant messages.
```

Comment:
Perhaps you may want to elaborate how the ui works with the parser module over here. The explanation could perhaps be improved by elaborating on the relation between the ui and the parser